twilight-model/src/gateway/payload/incoming/member_add.rs

```diff
@@ -1,9 +1,12 @@
-use crate::guild::Member;
+use crate::{
+    guild::Member,
+    id::{marker::GuildMarker, Id},
+};
 use serde::{Deserialize, Serialize};
 use std::ops::{Deref, DerefMut};
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct MemberAdd(pub Member);
+pub struct MemberAdd(pub Member, pub Id<GuildMarker>);
```

Comment:
My personal preference, might not be ideal so consider it yourself too.
```suggestion
pub struct MemberAdd {
    pub guild_id: Id<GuildMarker>,
    pub member: Member,
}
```